nova/tests/unit/scheduler/filters/test_shard_filter.py

```diff
@@ -63,174 +73,396 @@ def set_cache():
+            availability_zone='az-2',
+            instance_uuid=self.fake_build_req.instance_uuid,
+            flavor=fake_flavor.fake_flavor_obj(
+                mock.sentinel.ctx, expected_attrs=['extra_specs'],
+                name='m1'))
+
+        return list(self.filt_cls.filter_all(
+            [host1, host2, host3, host4, host5], spec_obj))
+
+    @mock.patch('nova.objects.AggregateList.get_by_host')
+    @mock.patch('nova.context.scatter_gather_skip_cell0')
+    @mock.patch('nova.objects.BuildRequest.get_by_instance_uuid')
+    @mock.patch('nova.context.get_admin_context')
+    def test_k8s_bypass_hana_flavors(self, get_context,
+                                     get_by_uuid,
+                                     gather_host,
+                                     get_aggrs):
+        gardener_cluster = 'kubernetes.io-cluster-shoot--garden--testCluster'
+        hana_flavor = fake_flavor.fake_flavor_obj(
+            mock.sentinel.ctx, expected_attrs=['extra_specs'],
+            id=1, name='hana_flavor1', memory_mb=256, vcpus=1, root_gb=1)
+        new_instance = fake_instance.fake_instance_obj(
+            get_context.return_value,
+            flavor=hana_flavor,
+            expected_attrs=['metadata'],
+            metadata={gardener_cluster: '1'})
+        build_req = objects.BuildRequest()
+        build_req.instance = new_instance
+        build_req.tags = objects.TagList()
+
+        get_by_uuid.return_value = build_req
+
+        self.filt_cls._PROJECT_SHARD_CACHE['baz'] = ['sharding_enabled',
+                                                     'vc-a-1']
+        agg1 = objects.Aggregate(id=1, name='vc-a-0', hosts=['host1'])
+        hana_agg = objects.Aggregate(id=1, name='vc-b-0',
+                                     hosts=['host2', 'host3'])
+
+        host1 = fakes.FakeHostState('host1', 'compute',
+                                    {'aggregates': [agg1]})
+        host2 = fakes.FakeHostState('host2', 'compute',
+                                    {'aggregates': [hana_agg]})
+        host3 = fakes.FakeHostState('host3', 'compute',
+                                    {'aggregates': [hana_agg]})
+        get_aggrs.return_value = [agg1, hana_agg]
 
+        spec_obj = objects.RequestSpec(
+            context=get_context.return_value, project_id='foo',
+            availability_zone='az-1',
+            instance_uuid=self.fake_build_req.instance_uuid,
+            flavor=fake_flavor.fake_flavor_obj(
+                mock.sentinel.ctx, expected_attrs=['extra_specs'],
+                name='hana_flavor1'))
+
+        result = list(self.filt_cls.filter_all([host1, host2, host3],
+                                               spec_obj))
+
+        gather_host.assert_not_called()
+        self.assertEqual(3, len(result))
+        self.assertEqual(result[0].host, 'host1')
+        self.assertEqual(result[1].host, 'host2')
+        self.assertEqual(result[2].host, 'host3')
```

Comment:
because there's a `host` property...

Reply:
Ah. How did I not see that? :D 
So it would have to be `self.assertEqual([r.host for r in result], ['host1', 'host2', 'host3'])`. Yeah, not sure if that's any better.